### DISCUSSION TITLE
Is it safe to use the latest version?

### DISCUSSION THREAD

**Discussion by 107ef363-c60f-46fb-a8bc-da8c5f4f6938** [actions/runner#1307] [Q&A]
I'm writing a script that registers a self-hosted runner. 

Part of the script deals with downloading the runner binary. I would prefer to not have to frequently update the runner version in this script, so I'm using `curl --silent "https://api.github.com/repos/actions/runner/releases/latest" | jq -r .tag_name` to get the latest release version. Currently this returns `v2.281.1`.

However I've noticed the instructions provided by the GitHub settings when adding a runner manually currently use `v2.280.3`. Is this a security measure? Should I be using that version (two behind latest) in my script as well or is it safe to stick with the latest release?

**Comment by rr-nick-tan** [comment]
I think no matter which version you install now, the runner itself has auto update mechanism that update to the latest released version.

**Comment by TingluoHuang** [comment]
https://docs.github.com/en/rest/reference/actions#list-runner-applications-for-a-repository

**Reply by 107ef363-c60f-46fb-a8bc-da8c5f4f6938** [reply]
@TingluoHuang Thanks, for pointing me to that documentation. Unfortunately there appears to be a bug with that endpoint right now (presumably low priority because it is rarely used) which makes it impossible to use (it returns an unhelpful error). I just opened an [issue](https://github.com/github/rest-api-description/issues/533) regarding this but I doubt it will get fixed soon. Is @rr-nick-tan correct that it automatically updates to the very latest release? If that is the case, I'll just fetch the latest version and use that from the start.
